microstack/review-single-node-quickstart/single-node-quickstart.md

```diff
@@ -0,0 +1,76 @@
+This tutorial shows how to install MicroStack in the simplest way possible. It will deploy an **OpenStack 2023.1** (Antelope) cloud.
+
+The cloud will only allow access to its VMs from the local host. To enable access from any host on your network, follow the [Single-node guided](/t/35765) tutorial instead.
+
+## Requirements
+
+You will need a single machine whose requirements are:
+
+- physical or virtual machine running Ubuntu 22.04 LTS
+- a multi-core amd64 processor ideally with 4+ cores
+- a minimum of 16 GiB of free memory
+
+[note type="caution"]
+**Caution:** Any change in IP address of the local host will be detrimental to the deployment. A virtual host will generally have a more stable address.
+[/note]
+
+## Deploy the cloud
+
+### Install the openstack snap
+
+Begin by installing the openstack snap:
+
+    sudo snap install openstack --channel 2023.1
+
+### Prepare the machine
+
+Sunbeam can generate a script to ensure that the machine has all of the required dependencies installed and is configured correctly for use in MicroStack - you can review this script using:
+
+    sunbeam prepare-node-script
+
+or the script can be directly executed in this way:
+
+    sunbeam prepare-node-script | bash -x && newgrp snap_daemon
+
+The script will ensure some software requirements are satisfied on the host. In particular, it will:
+
+* install `openssh-server` if it is not found
+* configure passwordless sudo for all commands for the current user (`NOPASSWD:ALL`)
```

Comment:
I haven't installed OpenStack to review the script, but does it (and if so should it) retain passwordless sudo for the user after everything's finished, or revert to requiring the password?

Reply:
I had the same question - it doesn't read like this part is happening inside a sandboxed environment

Reply:
It is sort of suggested that this is to be run in a VM and I was imagining that it would be somewhat ephemeral one that gets binned, but worth noting somehow, somewhere.

Reply:
It's great to get this concern. I brought it up a long while ago with the team but I'm going to do so again. I think that escalated privileges should be revoked once a node is added to the cluster, which is what needs the privileges.

Reply:
Filed an issue [here](https://bugs.launchpad.net/snap-openstack/+bug/2036811).